src/script/conversation/ConversationRepository.ts

```diff
@@ -184,8 +187,14 @@ export class ConversationRepository {
     // we register a client mismatch handler agains the message repository so that we can react to missing members
     // FIXME this should be temporary. In the near future we want the core to handle clients/mismatch/verification. So the webapp won't need this logic at all
     this.messageRepository.setClientMismatchHandler(async (mismatch, conversation, silent, consentType) => {
+      //we filter out self client id to omit it in mismatch check
+      const {userId, clientId} = this.core;
+      const selfClient = {domain: conversation?.domain, userId, clientId};
+      const filteredMissing = mismatch.missing && removeClientFromUserClientMap(mismatch.missing, selfClient);
+      const filteredMismatch = {...mismatch, missing: filteredMissing} as ClientMismatch | MessageSendingStatus;
```

Comment:
For some reason TS is not happy, even though we spread original object there and `mismatch.missing` has exact same type as `filteredMissing` 🤔

Reply:
interesting indeed 🤔